gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/service/tasks/LaunchConfigTask.java

```diff
@@ -148,10 +149,19 @@ public final FileCollection getClasspath() {
         return classpath;
     }
 
+    @InputFiles
+    public final FileCollection getJavaAgents() {
```

Comment:
I would use a `ConfigurableFileCollection` from Gradle. It is more idiomatic and allows you to avoid the setter. You could even use Gradle's magic property injection instead of defining a field:

```
@InputFiles
abstract ConfigurableFileCollection getJavaAgents();
```

Reply:
Should we update the classpath to do the same?

Reply:
ya we could honestly clean up all of the fields here to get rid of a bunch of boilerplate. I didn't want to slam you with unrelated changes though

Reply:
No worries, I implemented this one by pattern matching against the classpath config.